lib/widgets/dialog.dart

```diff
@@ -15,6 +16,30 @@ Widget _dialogActionText(String text) {
   );
 }
 
+/// Sets the dialog action to be platform appropriate
+/// by displaying a [CupertinoDialogAction] for IOS platforms
+/// and a regular [TextButton] otherwise.
+Widget _adaptiveAction(
+      {required BuildContext context,
+      required VoidCallback onPressed,
+      required Widget child}) {
+    final ThemeData theme = Theme.of(context);
+    switch (theme.platform) {
+      case TargetPlatform.android:
+        return TextButton(onPressed: onPressed, child: child);
+      case TargetPlatform.fuchsia:
+      return TextButton(onPressed: onPressed, child: child);
+      case TargetPlatform.linux:
+        return TextButton(onPressed: onPressed, child: child);
+      case TargetPlatform.windows:
+        return TextButton(onPressed: onPressed, child: child);
+      case TargetPlatform.iOS:
+        return CupertinoDialogAction(onPressed: onPressed, child: child);
+      case TargetPlatform.macOS:
+        return CupertinoDialogAction(onPressed: onPressed, child: child);
+    }
+  }
```

Comment:
This can be tightened up in a few ways:

- Use empty `case`s to fall through. From the [doc on Dart `switch` statements](https://dart.dev/language/branches#switch-statements):
  
  > https://dart.dev/language/branches#switch-statements
- Use `defaultTargetPlatform` instead of passing through `context`. When the app is run on a device, `defaultTargetPlatform` will match the platform (iOS or Android). When we need to simulate a specific platform in tests, we set `defaultTargetPlatform`; search for `debugDefaultTargetPlatformOverride` for how we do this.
- More concise dartdoc

Also, because the logic in `_dialogActionText` fits and is recommended for the Material-style dialog, let's only apply it on Android. This helper is a fine place for that conditional logic; how about changing its interface so it takes a String instead of a Widget, and applies `_dialogActionText` in the Android branch. In my proposal below, I've also given `_dialogActionText` an appropriately specific name, `_materialDialogActionText`. That helper could even be inlined, perhaps in a followup [NFC](https://github.com/zulip/zulip-mobile/blob/main/docs/glossary.md#user-content-nfc) commit.

So, putting all that together:

```suggestion
/// A platform-appropriate action for [AlertDialog.adaptive]'s [actions] param.
Widget _adaptiveAction({required VoidCallback onPressed, required String text}) {
  switch (defaultTargetPlatform) {
    case TargetPlatform.android:
    case TargetPlatform.fuchsia:
    case TargetPlatform.linux:
    case TargetPlatform.windows:
      return TextButton(onPressed: onPressed, child: _materialDialogActionText(text));
    case TargetPlatform.iOS:
    case TargetPlatform.macOS:
      return CupertinoDialogAction(onPressed: onPressed, child: Text(text));
  }
}
```

Reply:
Is this the idea for inlining the `_materialDialogActionText`?
```dart
return TextButton(onPressed: onPressed, child: Text(text, textAlign: TextAlign.end));
```
will update the commit message of the latest commit as well as per the guidelines